core/src/main/scala/org/broadinstitute/dsp/workbench/welder/model.scala

```diff
@@ -1,9 +1,56 @@
 package org.broadinstitute.dsp.workbench.welder
 
-import io.circe.Decoder
+import java.time.Instant
+
+import cats.implicits._
+import io.circe.{Decoder, Encoder}
+import ca.mrvisser.sealerate
+import org.broadinstitute.dsde.workbench.google2.GcsBlobName
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.GcsBucketName
+import org.http4s.Uri
 
 final case class LocalObjectPath(asString: String) extends AnyVal
+sealed abstract class SyncStatus extends Product with Serializable
+object SyncStatus {
+  // crc32 match
+  final case object Live extends SyncStatus {
+    override def toString: String = "LIVE"
+  }
+  // crc32 mismatch
+  final case object Desynchronized extends SyncStatus {
+    override def toString: String = "DESYNCHRONIZED"
+  }
+  // deleted in gcs. (object exists in storagelinks config file but not in in gcs)
+  final case object DeletedRemote extends SyncStatus {
+    override def toString: String = "DELETED_REMOTE"
+  }
+
+  val stringToSyncStatus: Set[SyncStatus] = sealerate.values[SyncStatus]
+}
+final case class BucketNameAndObjectName(bucketName: GcsBucketName, blobName: GcsBlobName) {
+  override def toString: String = s"${bucketName.value}/${blobName.value}"
+}
 
 object JsonCodec {
   implicit val localObjectPathDecoder: Decoder[LocalObjectPath] = Decoder.decodeString.map(LocalObjectPath)
-}
\ No newline at end of file
+  implicit val localObjectPathEncoder: Encoder[LocalObjectPath] = Encoder.encodeString.contramap(_.asString)
```

Comment:
Was swagger codegen not an option? It handles all of these encoding details for you.

Reply:
I don't know if swagger codegen will give me circe encoder/decoder. But without swagger codegen, it's pretty easy just to generate encoder/decoder (all you need is `import io.circe.generic.auto._`), but I prefer to write encoder/decoder manually because it gives us more control, and we can bake more logic into it for validating json, and better error messages